### ISSUE TITLE
RuntimeBinderException: Cannot convert null to 'bool' because it is a non-nullable value type when trying to edit ContentItem with WidgetList that has widgets associated with it after upgrade from 1.4.0 to 1.5.0.

### ISSUE THREAD

**Issue by Andrew120219** [OrchardCMS/OrchardCore#12948]
Small back story that led me to find this. There was an issue with the Sitemaps feature that was not showing the site root in sitemap.xml. It instead would show the homepage as the permalink (IE www.example.com/homepage and not www.example.com). I read that issue was fixed with 1.5.0 so I upgraded my project to use 1.5.0. 

After the upgrade I cleared the cache of the sitemap and then went to sitemap.xml to see if it fixed the issue and it was still showing the same.  I made a different page and set that to be the homepage. After setting the new homepage, I saw the expected functionality and the site root was listed in my sitemap.xml.  So, I was going to try and set it back to the original homepage. I tried to open and edit the homepage content item and was met with the following error: 

```
RuntimeBinderException: Cannot convert null to 'bool' because it is a non-nullable value type
CallSite.Target(Closure , CallSite , object )
System.Dynamic.UpdateDelegates.UpdateAndExecute1<T0, TRet>(CallSite site, T0 arg0)
AspNetCoreGeneratedDocument.Views_ContentCard_WidgetsListPart_Edit.ExecuteAsync()
Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageCoreAsync(IRazorPage page, ViewContext context)
Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageAsync(IRazorPage page, ViewContext context, bool invokeViewStarts)
Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderAsync(ViewContext context)
Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper.RenderPartialCoreAsync(string partialViewName, object model, ViewDataDictionary viewData, TextWriter writer)
Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper.PartialAsync(string partialViewName, object model, ViewDataDictionary viewData)
OrchardCore.DisplayManagement.Implementation.DefaultHtmlDisplay.<ProcessAsync>g__Awaited|11_0(Task<IHtmlContent> task)
OrchardCore.DisplayManagement.Implementation.DefaultHtmlDisplay.ExecuteAsync(DisplayContext context)
OrchardCore.DisplayManagement.Implementation.DefaultHtmlDisplay.ExecuteAsync(DisplayContext context)
```

The original homepage will render if I visit it's Url, www.example.com/homepage, but if I try to access it in the admin to edit anything about it, I get the above error. 

I started doing some troubleshooting and narrowed it down to something with the widget list. I created a new content type and added my widgets to a flow instead of using the widget list. It worked and I was able to open the content item edit view.

If I set the version back to 1.4.0, I am able to open the edit screen for the original homepage content item in the admin. I am not sure what is going on. Any help will be greatly appreciated.

[StackTrace_WidgetListError.txt](https://github.com/OrchardCMS/OrchardCore/files/10222029/StackTrace_WidgetListError.txt)

**Comment by MikeAlhayek** [comment]
@Andrew120219 I think this issue is related to #12938 

 It should have been fixed by PR #12939 if you want to test it out, please use OC 1.6 preview.

**Comment by MikeAlhayek** [comment]
@Andrew120219 is this still an issue?
